tasks.go

```diff
@@ -137,19 +139,98 @@ func GetAllTasks(runner Runner, basepath string) <-chan Task {
 			defer wg.Done()
 			tasks <- GetOcAdmDiagnosticsTask(runner)
 		}()
+		wg.Wait()
+	}()
+	return tasks
+}
+
+func RunAllAnalysisTasks(runner Runner, path string, workers int) {
+	start := time.Now()
+
+	checkResults := make(chan CheckResults)
+	tasks := GetAllAnalysisTasks(runner, path, checkResults)
+	results := make(chan error)
 
+	// Start worker goroutines to run tasks concurrently.
+	var wg sync.WaitGroup
+	wg.Add(workers)
+	for i := 0; i < workers; i++ {
+		go func() {
+			defer wg.Done()
+			for task := range tasks {
+				results <- task()
+			}
+		}()
+	}
+	// Wait for all workers to terminate, then close the results channel to
+	// communicate that no more results will be sent.
+	go func() {
+		go func() {
+			analysisResults := map[string][]Result{}
+			filepath := filepath.Join(path, "analysis.json")
+			err := os.MkdirAll(path, 0770)
+			if err != nil {
+				results <- err
+				return
+			}
+			analysisFile, err := os.Create(filepath)
+			if err != nil {
+				results <- err
+				return
+			}
+			defer analysisFile.Close()
+
+			for result := range checkResults {
+				analysisResults[result.Scope] = result.Results
+				output, err := json.MarshalIndent(analysisResults, "", "    ")
+				if err != nil {
+					results <- err
+				}
+				analysisFile.Write(output)
+			}
+		}()
 		wg.Wait()
```

Comment:
this seems a bit complex.  do we need the nested go func, not sure we do. Seems like it could even be a separate function?